mitosheet/mitosheet/step_performers/graph_steps/graph.py

```diff
@@ -102,8 +105,21 @@ def execute(cls, prev_state: State, params: Dict[str, Any]) -> Tuple[State, Opti
         y_axis_column_ids = graph_creation["y_axis_column_ids"] if graph_creation["y_axis_column_ids"] is not None else []
         y_axis_column_headers = prev_state.column_ids.get_column_headers_by_ids(sheet_index, y_axis_column_ids)
 
-        # Validate optional parameters 
-        color_column_header = prev_state.column_ids.get_column_header_by_id(sheet_index, graph_creation["color"]) if 'color' in graph_creation.keys() else None
+        # Validate optional parameters that are available for all graph types
+        color_column_header = prev_state.column_ids.get_column_header_by_id(sheet_index, graph_creation["color"]) if 'color' in graph_creation.keys() and graph_creation["color"] in prev_state.column_ids.column_id_to_column_header[sheet_index].keys() else None
+        facet_col_column_header = prev_state.column_ids.get_column_header_by_id(sheet_index, graph_creation["facet_col_column_id"]) if 'facet_col_column_id' in graph_creation.keys() and graph_creation["facet_col_column_id"] in prev_state.column_ids.column_id_to_column_header[sheet_index].keys() else None
+        facet_row_column_header = prev_state.column_ids.get_column_header_by_id(sheet_index, graph_creation["facet_row_column_id"]) if 'facet_row_column_id' in graph_creation.keys() and graph_creation["facet_row_column_id"] in prev_state.column_ids.column_id_to_column_header[sheet_index].keys() else None
+        facet_col_wrap = int(graph_creation['facet_col_wrap']) if 'facet_col_wrap' in graph_creation.keys() else None
```

Comment:
Here's a helpful bit of syntax:
```
graph_creation.get('facet_col_wrap', None)
```
The second parameter is the default if the key is not there. But it doesn't work because of the cast. 

Is there anyway we can avoid the cast by making this as an integer on the frontend? Or is this hard for some reason?

Reply:
Fixed up so all casting is on the frontend